protos/flyteidl/admin/description_entity.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package flyteidl.admin;
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin";
+
+import "flyteidl/core/identifier.proto";
+import "flyteidl/admin/common.proto";
+
+// DescriptionEntity contains detailed description for the task/workflow/launch plan.
+// Documentation could provide insight into the algorithms, business use case, etc.
+message DescriptionEntity {
+  // One-liner overview of the entity.
+  string short_description = 1;
+  // Full user description with formatting preserved.
+  LongDescription long_description = 2;
+  // Optional link to source code used to define this entity.
+  SourceCode source_code = 5;
+}
+
+// Full user description with formatting preserved. This can be rendered
+// by clients, such as the console or command line tools with in-tact
+// formatting.
+message LongDescription {
+  oneof content {
+    // long description - no more than 4KB
+    string value = 1;
+    // if the description sizes exceed some threshold we can offload the entire
+    // description proto altogether to an external data store, like S3 rather than store inline in the db
+    string uri = 2;
+  }
+
+  enum DescriptionFormat {
+    UNKNOWN = 0;
+    MARKDOWN = 1;
+    HTML = 2;
+    // python default documentation - comments is rst
+    RST = 3;
+  }
+  // Format of the long description
+  DescriptionFormat format = 3;
+  // Optional link to an icon for the entity
+  string icon_link = 4;
+}
+
+// Link to source code used to define this entity
+message SourceCode {
+  string link = 5;
```

Comment:
```suggestion
  string link = 1;
```

Reply:
I didn't add too much metadata for now in the source code because It's hard to revert it. 
yup, I can add it also if we want it.

Reply:
sounds good, we can iterate on this